src/FluxCalculator.jl

```diff
@@ -91,4 +100,296 @@ function calculate_surface_air_density(atmos_sim::Interfacer.AtmosModelSimulatio
+"""
+function surface_inputs(
+    ::BulkScheme,
+    thermo_state_sfc,
+    thermo_state_int,
+    uₕ_int,
+    z_int,
+    z_sfc,
+    z0b,
+    z0m,
+    Ch,
+    Cd,
+    beta,
+    gustiness,
+)
+    FT = Spaces.undertype(axes(z_sfc))
+
+    # wrap state values
+    return @. SF.Coefficients(
+        SF.InteriorValues(z_int, uₕ_int, thermo_state_int), # state_in                              # state_sfc
+        SF.SurfaceValues(                                  # state_sfc
+            z_sfc,
+            StaticArrays.SVector(FT(0), FT(0)),
+            thermo_state_sfc,
+        ),
+        Cd,                                     # Cd
+        Ch,                                     # Ch
+        z0m,                                    # roughness_momentum
+        z0b,                                    # roughness_buoyancy
+        gustiness,                              # gustiness
+        beta,                                   # beta
+    )
+end
+function surface_inputs(
+    ::MoninObukhovScheme,
+    thermo_state_sfc,
+    thermo_state_int,
+    uₕ_int,
+    z_int,
+    z_sfc,
+    z0b,
+    z0m,
+    Ch,
+    Cd,
+    beta,
+    gustiness,
+)
+    FT = Spaces.undertype(axes(z_sfc))
+
+    # wrap state values
+
+    return @. SF.ValuesOnly(
+        SF.InteriorValues(z_int, uₕ_int, thermo_state_int), # state_in
+        SF.SurfaceValues(                                  # state_sfc
+            z_sfc,
+            StaticArrays.SVector(FT(0), FT(0)),
+            thermo_state_sfc,
+        ),
+        z0m,                                    # z0m
+        z0b,                                    # z0b
+        gustiness,                             # gustiness
+        beta,                                   # beta
+    )
+
+end
+
+"""
+    surface_thermo_state(sim::Interfacer.SurfaceModelSimulation, thermo_params::TD.Parameters.ThermodynamicsParameters, thermo_state_int, colidx::Fields.ColumnIndex)
+
+Returns the surface parameters for the surface model simulation `sim`.
+"""
+function surface_thermo_state(
+    sim::Interfacer.SurfaceModelSimulation,
+    thermo_params::TD.Parameters.ThermodynamicsParameters,
+    thermo_state_int,
+    colidx::Fields.ColumnIndex,
+)
+    @warn("Simulation " * Interfacer.name(sim) * " uses the default thermo (saturated) surface state", maxlog = 10)
+    T_sfc = Interfacer.get_field(sim, Val(:surface_temperature), colidx) #
+    ρ_sfc = extrapolate_ρ_to_sfc.(thermo_params, thermo_state_int, T_sfc) # ideally the # calculate elsewhere, here just getter...
+    q_sfc = TD.q_vap_saturation_generic.(thermo_params, T_sfc, ρ_sfc, TD.Liquid()) # default: saturated liquid surface
+    @. TD.PhaseEquil_ρTq.(thermo_params, ρ_sfc, T_sfc, q_sfc)
+end
+
+# TODO: (an equivalent of) this function also lives in Atmos and Land - should move to general utilities
+"""
+    extrapolate_ρ_to_sfc(thermo_params, ts_int, T_sfc)
+
+Uses the ideal gas law and hydrostatic balance to extrapolate for surface density pointwise.
```

Comment:
Why is this pointwise instead of just broadcasting the operations?

Reply:
Actually it is being broadcasted, maybe a misleading comment - it's been reworded.